### PR TITLE
Devel/eapol manager 2

### PR DESCRIPTION
**Pull request by metelik**
- Bugfixing Nerves.Network.EAPoLManager.start/1 call
- Correcting and expanding UTs
- Adding forward of :"CTRL-EVENT-TERMINATING"
- Adding handling of terminate/2 Nerves.Network.EAPoLManager and associated event

### CODE REVIEW

**Review by PeterJeaiem — APPROVED** [review]

File: lib/nerves_network/eapol_manager.ex [hunk 1]
```diff
@@ -223,7 +224,8 @@ defmodule Nerves.Network.EAPoLManager do
       pid = start_wpa_supervisor(state)
 
       :ok = registry_register(Nerves.WpaSupplicant, state.ifname)
-      %{%{state | supplicant_port: port} | wpa_pid: pid}
+
+       %{%{state | supplicant_port: port} | wpa_pid: pid}
```
Comment (david-naughton) [inline lib/nerves_network/eapol_manager.ex]: This should probably be %{state | supplicant_port: port, wpa_pid: pid}

File: lib/nerves_network/eapol_manager.ex [hunk 2]
```diff
@@ -335,6 +335,15 @@ defmodule Nerves.Network.EAPoLManager do
     {:noreply, s}
   end
 
+  # Hnadling events like: {Nerves.WpaSupplicant, :"CTRL-EVENT-TERMINATING", %{ifname: "eth0"}} and
```
Comment (david-naughton) [inline lib/nerves_network/eapol_manager.ex]: Typo in 'Handling'

**Review by david-naughton — APPROVED** [review]

File: lib/nerves_network/eapol_manager.ex (hunk 1 above)
Comment (adrigonzo) [reply lib/nerves_network/eapol_manager.ex]: both work, I think, but David's way does look cleaner.

File: lib/nerves_network/if_supervisor.ex [hunk 3]
```diff
@@ -247,8 +253,12 @@ defmodule Nerves.Network.IFSupervisor do
 
     managers_v4 = ipv4_managers(settings)
     managers_v6 = ipv6_managers(settings)
+    eapol       = eapol_managers(settings)
+
+    managers =
```
Comment (adrigonzo) [inline lib/nerves_network/if_supervisor.ex:258]: if you don't have any debug messages in between, I'd go back to previous version without the `managers` variable in between.

File: test/nerves_eapolmanager_test.exs [hunk 4]
```diff
@@ -7,10 +7,50 @@ defmodule Nerves.Network.EAPoLManagerTest do
...
+    :timer.sleep 1111
+
+    #Nerves.Network.EAPoLManager.start("eth0", setup)
+
+    :ok
+  end
+
+  defp list_wpa_processes() do
+      x = System.cmd "ps", ["-C", "wpa_supplicant"]
+      IO.puts "+++ ps = #{inspect x}"
```
Comment (adrigonzo) [inline test/nerves_eapolmanager_test.exs]: formatting

**Review by adrigonzo — APPROVED** [review]

File: lib/nerves_network/eapol_manager.ex (hunk 1 above)
Comment (metelik) [reply lib/nerves_network/eapol_manager.ex]: can one update two items at once?

Comment (alanj853) [reply lib/nerves_network/eapol_manager.ex]: As long as the keys existed beforehand, then Yes. If not, you could do this 
```
state
|> Map.put(:supplicant_port, port)
|> Map.put(:wpa_pid, pid)
```

File: lib/nerves_network/eapol_manager.ex [hunk 5]
```diff
@@ -377,6 +394,15 @@ defmodule Nerves.Network.EAPoLManager do
     iex> is_map(retval)
     true
 
+    iex> setup = %{
```
Comment (alanj853) [inline lib/nerves_network/eapol_manager.ex:397]: This should be an in-file UT, not a doctest.

File: lib/nerves_network/eapol_manager.ex [hunk 6]
```diff
@@ -476,6 +502,15 @@ defmodule Nerves.Network.EAPoLManager do
 
   ## Examples
 
+  #iex> setup = %{
```
Comment (alanj853) [inline lib/nerves_network/eapol_manager.ex:505]: This should be an in-file UT, not a doctest.

File: test/nerves_eapolmanager_test.exs [hunk 7]
```diff
@@ -22,18 +62,100 @@ defmodule Nerves.Network.EAPoLManagerTest do
...
+     }
+    retval = Nerves.Network.EAPoLManager.start("eth0", setup)
+    assert is_map(retval)
 
-    # Context is not updated here
-    :ok
+  end
+
+  test "start_2" do
+    IO.puts "+++ Test start_2"
```
Comment (alanj853) [inline test/nerves_eapolmanager_test.exs]: This is not needed, --trace can be added to the test command to provide the same information. You have a few instances of this, so I'm just going to mention it here.

**Review by alanj853 — COMMENTED** [review]
Looks good, but needs a few touch ups before merging.

File: lib/nerves_network/eapol_manager.ex (hunk 1 above)
Comment (metelik) [reply lib/nerves_network/eapol_manager.ex]: re-wroked.

File: lib/nerves_network/eapol_manager.ex (hunk 2 above)
Comment (metelik) [reply lib/nerves_network/eapol_manager.ex]: corrected.

File: lib/nerves_network/eapol_manager.ex (hunk 6 above)
Comment (metelik) [reply lib/nerves_network/eapol_manager.ex:505]: The provided example illustrates the use of the API and that's the goal

File: lib/nerves_network/if_supervisor.ex (hunk 3 above)
Comment (metelik) [reply lib/nerves_network/if_supervisor.ex:258]: changed that :)

File: test/nerves_eapolmanager_test.exs (hunk 7 above)
Comment (metelik) [reply test/nerves_eapolmanager_test.exs]: Cool - thanks @alanj853 - learning everyday :)

File: test/nerves_eapolmanager_test.exs (hunk 4 above)
Comment (metelik) [reply test/nerves_eapolmanager_test.exs]: ran formatter

File: lib/nerves_network/eapol_manager.ex (hunk 5 above)
Comment (metelik) [reply lib/nerves_network/eapol_manager.ex:397]: I disagree, the provided example illustrates the use of the API and that's the goal

**Pull request merged**